### PR TITLE
Ensure we clean up model persistence table when model or DB is permanently deleted

### LINKED ISSUE #48664: Deleting a model doesn't remove it from database.
### Describe the bug

Deleting a model doesn't remove from database.

### To Reproduce

1. Go to View archive
2. Delete a model
3. Go to the database side
4. The model is still persisted

### Expected behavior

It should be removed from the database

### Logs

_No response_

### Information about your Metabase installation

Metabase version: v0.50.29
Metabase hosting environment: Jar-file
Metabase internal database : MySQL

### Severity

Unable to clean the database, which consumes disk space and leads to database clutter.

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by noahmoss**
Resolves https://github.com/metabase/metabase/issues/48664

* Drops the NOT NULL constraint on `persisted_info.card_id`
* Drops and recreates the FK constraint with a `SET NULL` deletion policy
* Updates the persist refresh task to prune tables which have `null` card IDs

This ensures that if you permanently delete a model, it won't delete the `persisted_info` record immediately, but rather keep it around until the next time the refresh task runs so that the underlying table can be pruned.

I'm explicitly _not_ handling past cases where a model persistence table has gotten orphaned due to deletion of the model. We don't have the persistence record available anymore, so it's not trivial—maybe we could build tooling to detect these in sync and surface them to the admin or something, but I think that's out of scope for this bug fix.

### CODE REVIEW

**Comment by noahmoss** [comment]
* **#49112** <a href="https://app.graphite.dev/github/pr/metabase/metabase/49112?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `master`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @noahmoss and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: resources/migrations/001_update_migrations.yaml [hunk 1]
```diff
@@ -8719,7 +8719,7 @@ databaseChangeLog:
         - not:
             - foreignKeyConstraintExists:
                 foreignKeyName: fk_pulse_channel_channel_id
-                foreignKyeTableName: pulse_channel
+                foreignKeyTableName: pulse_channel
```
Comment (dpsutton) [inline resources/migrations/001_update_migrations.yaml:8722]: woah. this is kinda nuts

Comment (noahmoss) [reply resources/migrations/001_update_migrations.yaml:8722]: yeah...I'm not sure what the existing behavior is with that typo

Comment (noahmoss) [reply resources/migrations/001_update_migrations.yaml:8722]: actually, these typos were just in preconditions, so it shouldn't have affected the actual migration logic. but I presume the preconditions wouldn't have been enforced properly

Comment (escherize) [reply resources/migrations/001_update_migrations.yaml:8722]: Wow, yea this could be some evidence that we should improve linting on this file.

Comment (escherize) [reply resources/migrations/001_update_migrations.yaml:8722]: Found another here: https://github.com/metabase/metabase/blob/master/resources/migrations/001_update_migrations.yaml#L9760

~Not sure if we should actually fix it here.~ @noahmoss fixed it in this pr 😎

File: resources/migrations/001_update_migrations.yaml [hunk 2]
```diff
@@ -9757,7 +9815,7 @@ databaseChangeLog:
         - not:
           - foreignKeyConstraintExists:
               foreignKeyName: fk_notification_handler_template_id
-              foreignKyeTableName: notification_handler
+              foreignKeyTableName: notification_handler
```
Comment (escherize) [inline resources/migrations/001_update_migrations.yaml:9818]: thanks

File: src/metabase/task/persist_refresh.clj [hunk 3]
```diff
@@ -154,9 +152,10 @@
 (defn- deletable-models
   "Returns persisted info records that can be unpersisted. Will select records that have moved into a deletable state
   after a sufficient delay to ensure no queries are running against them and to allow changing mind. Also selects
-  persisted info records pointing to cards that are no longer models and archived cards/models."
+  persisted info records pointing to cards that are no longer models, archived cards/models, and records where the corresponding
+  card or database has been permanently deleted."
```
Comment (escherize) [inline src/metabase/task/persist_refresh.clj:156]: I kept reading this as the verb `records`.

```suggestion
  persisted info records pointing to cards that are no longer models, archived cards/models, and all records where the corresponding
  card or database has been permanently deleted."
```

**Review by escherize — APPROVED** [review]
LGTM

**Pull request merged**